### DISCUSSION TITLE
How to access FieldError object by index

### DISCUSSION THREAD

**Discussion by tylerbunting** [react-hook-form/react-hook-form#12034] [Q&A] [answered]
Hi there, firstly I have never made a discussion on GitHub before so apologies if I use any incorrect terminology.

So I am creating a form to create a template. These templates can have multiple cards so all components are stored in a field called cardComponents which utilizes useFieldArray. Within these cards there is an optional field for buttons which is also using useFieldArray. The input fields for these buttons are stored in a different react component. I pass this component the errors like so:

`<CreateTemplateButtonFields 
    cardIndex={cardIndex} 
    control={componentsControl} 
    register={componentsRegister}
    sources={sources}
    errors={componentsErrors.cardComponents?.[cardIndex]?.buttons}
    buttonConfig={buttonField}
  />`
  
The value being passed to the errors prop has type: FieldError | undefined so I pass the props like this:

`interface CreateTemplateButtonFieldsProps {
  cardIndex: number;
  control: Control<ComponentsFormValues>;
  register: UseFormRegister<ComponentsFormValues>;
  sources: MessageSource[];
  buttonConfig?: CreateTemplateComponent;
  errors?: FieldError;
}`
  
I am trying to access the values of these errors within the components like so:
  
`<TextField
    fullWidth 
    label="Title"
    error={Boolean(props.errors?.[index]?.title)}
    helperText={props.errors?.[index]?.title.message}
    {...props.register(`cardComponents.${props.cardIndex}.buttons.${index}.title` as const)}
  />`
  
However, I am getting the following error underneath props.errors?.[index]: 
    
Element implicitly has an 'any' type because expression of type 'number' can't be used to index type 'FieldError'.
No index signature with a parameter of type 'number' was found on type 'FieldError'.

When I breakpoint the props.errors or console.log it, the value printed is an array. 

How do I access these values by index?

My button fields include: title, type and payload which should all be strings.

Thank you for the help.

**Comment by tatsmaki** [comment]
Would be nice to have code-sandbox to reproduce your case.

**Reply by tylerbunting** [reply]
Give me 5 minutes ill get one linked in here.

**Comment by tylerbunting** [comment, marked as answer]
Closing as realised error was to do with field type being either string | object.
